src/Eurofurence.App.Server.Services/PushNotifications/FirebaseChannelManager.cs

```diff
@@ -206,17 +206,27 @@ public async Task RegisterDeviceAsync(
             DeviceType type,
             CancellationToken cancellationToken = default)
         {
-            if (await _deviceService.FindAll(a => a.DeviceToken == deviceToken).AnyAsync(cancellationToken))
+            var existing = await _deviceService
+                .FindAll(a => a.DeviceToken == deviceToken)
+                .AsTracking()
+                .ToListAsync(cancellationToken);
+
+            if (existing.Count > 0)
             {
-                return;
+                foreach (var identity in existing)
+                {
+                    identity.IdentityId = identityId;
+                }
```

Comment:
Since we're modifying tracked entities, we should probably `await _appDbContext.SaveChangesAsync(cancellationToken);` to write those changes to the db?